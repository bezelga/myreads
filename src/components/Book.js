import React, {Component} from 'react'
import PropTypes from 'prop-types';
import Changer from './Changer'

class Book extends Component {
  render() {
    return(
    <div className="book">
      <div className="book-top">
        <div className="book-cover" style={{
           width: 128,
           height: 193,
           backgroundImage: `url(${this.props.coverURL})`
          }}>
        </div>
        <Changer currentBookshelf={this.props.shelf} changeBookStateFunction={this.props.changeBookStateFunction(this.props.title)}/>
      </div>
      <div className="book-title">{this.props.title}</div>
      <div className="book-authors">{this.props.authors}</div>
    </div>
  )}
}

Book.propTypes = {
  coverURL: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  authors: PropTypes.string.isRequired,
  shelf: PropTypes.string.isRequired
}

export default Book
