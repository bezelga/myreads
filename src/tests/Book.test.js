import React from 'react'
import Book from '../components/Book'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'

const changeBookStateFunction = () => () => {}
 const bookSample = <Book changeBookStateFunction={changeBookStateFunction} title="MyBook" id="123" shelf="read"></Book>

it('matches the book contract with final snapshot', () => {
  const component = renderer.create(
    bookSample
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Book changes favorite state after toggle', () => {
  const book = shallow(bookSample)

  book.find('.book-favorite').first().simulate('click', { preventDefault() {} })

  expect(book.find('.book-unfavorite').text()).toEqual('unfavorite');

  book.find('.book-unfavorite').first().simulate('click', { preventDefault() {} })

  expect(book.find('.book-favorite').text()).toEqual('make favorite');
});
