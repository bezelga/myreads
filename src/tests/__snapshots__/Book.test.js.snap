// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the book contract with final snapshot 1`] = `
<div
  className="book"
>
  <a
    className="book-favorite"
    href="/fav"
    onClick={[Function]}
  >
    make favorite
  </a>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url(undefined)",
          "height": 193,
          "width": 128,
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <select
        defaultValue="read"
        onChange={[Function]}
      >
        <option
          disabled={true}
          value="move"
        >
          Move to...
        </option>
        <option
          value="currentlyReading"
        >
          Currently Reading
        </option>
        <option
          value="wantToRead"
        >
          Want to Read
        </option>
        <option
          value="read"
        >
          Read
        </option>
        <option
          value="none"
        >
          None
        </option>
      </select>
    </div>
  </div>
  <div
    className="book-title"
  >
    MyBook
  </div>
  <div
    className="book-authors"
  >
    
  </div>
</div>
`;
